spec/requests/devise/registrations/user_signup_spec.rb

```diff
@@ -0,0 +1,177 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples 'external user not created' do
+  it { expect { request }.to change(User, :count).by(0) }
+  it { expect { request }.to change(ExternalUser, :count).by(0) }
+  it { expect { request }.to change(Provider, :count).by(0) }
+end
+
+RSpec.shared_examples 'external user created' do
+  it { expect { request }.to change(User, :count).by(1) }
+  it { expect { request }.to change(ExternalUser, :count).by(1) }
+  it { expect { request }.to change(Provider, :count).by(1) }
+end
+
+RSpec.describe 'User sign up', type: :request do
+  include Capybara::RSpecMatchers
+
+  describe 'GET #new' do
+    subject(:request) { get new_user_registration_path }
+
+    context 'when NOT on api-sandbox' do
+      around do |example|
+        with_env('production') do
```

Comment:
This (copied from `spec/controllers/external_users/registrations_controller_spec.rb`) is causing problems with the dependabot update PR for `database_cleaner`. Version 2.0 of that gem introduced a safeguard to prevent database modifications from happening if the environment is detected to be production.

If it is at all possible (and simple) to do this another way it would be great, but if not this isn't a blocker as there are other instances of this issue elsewhere anyway.

Reply:
this only changes the `ENV['ENV']` rather than `ENV['RAILS_ENV']` environment variable (which is not a typical rails app env var) but I will have a look. it can use `staging` as a value in any event, with no impact to tests or their underlying intention.